bitswap/server/internal/decision/engine.go

```diff
@@ -699,7 +722,10 @@ func (e *Engine) MessageReceived(ctx context.Context, p peer.ID, m bsmsg.BitSwap
 			continue
 		}
 
-		e.peerLedger.Wants(p, entry.Entry)
+		if !e.noPeerLedger {
+			e.peerLedger.Wants(p, entry.Entry)
+		}
+
```

Comment:
ℹ️ This should be enough as the only place to use `e.noPeerLedger`. All other operations will then return empty slices/0 or be no-ops.